project/dexter-peer-review/src/java/com/samsung/sec/dexter/peerreview/plugin/DexterPeerReviewPlugin.java

```diff
@@ -0,0 +1,311 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.samsung.sec.dexter.peerreview.plugin;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+
+import com.google.common.collect.Lists;
+import com.samsung.sec.dexter.core.analyzer.AnalysisConfig;
+import com.samsung.sec.dexter.core.analyzer.AnalysisEntityFactory;
+import com.samsung.sec.dexter.core.analyzer.AnalysisResult;
+import com.samsung.sec.dexter.core.analyzer.IAnalysisEntityFactory;
+import com.samsung.sec.dexter.core.checker.CheckerConfig;
+import com.samsung.sec.dexter.core.config.DexterConfig;
+import com.samsung.sec.dexter.core.config.DexterConfig.LANGUAGE;
+import com.samsung.sec.dexter.core.defect.Defect;
+import com.samsung.sec.dexter.core.defect.Occurence;
+import com.samsung.sec.dexter.core.plugin.IDexterPlugin;
+import com.samsung.sec.dexter.core.plugin.PluginDescription;
+import com.samsung.sec.dexter.core.plugin.PluginVersion;
+
+import net.xeoh.plugins.base.annotations.PluginImplementation;
+
+@PluginImplementation
+public class DexterPeerReviewPlugin implements IDexterPlugin {
+	private final static Logger logger = Logger.getLogger(DexterPeerReviewPlugin.class);
+
+	public final static String PLUGIN_NAME = "dexter-peerreview";
+	public final static PluginVersion PLUGIN_VERSION = new PluginVersion("0.0.1");
+	private static PluginDescription PLUGIN_DESCRIPTION;
+
+	private IAnalysisEntityFactory analysisEntityFactory = new AnalysisEntityFactory();
+
+	final String criticalRegExp = "(?i).*\\[.*cri.*\\] | (?i).*\\[.*critical.*\\]";
+	final String majorRegExp = "(?i).*\\[.*maj.*\\] | (?i).*\\[.*major.*\\]";
+	final String simpleRegExp = ".*(DPR:)";
+
+	public DexterPeerReviewPlugin() {
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void destroy() {
+
+	}
+
+	@Override
+	public void handleDexterHomeChanged(String oldPath, String newPath) {
+
+	}
+
+	@Override
+	public PluginDescription getDexterPluginDescription() {
+		if (PLUGIN_DESCRIPTION == null) {
+			PLUGIN_DESCRIPTION = new PluginDescription("Samsung Electroincs", PLUGIN_NAME, PLUGIN_VERSION,
+					DexterConfig.LANGUAGE.CPP, "");
+		}
+
+		return PLUGIN_DESCRIPTION;
+	}
+
+	@Override
+	public void setCheckerConfig(CheckerConfig cc) {
+
+	}
+
+	@Override
+	public CheckerConfig getCheckerConfig() {
+		return null;
+	}
+
+	static int[] createNewArray(int[] oldArray, int maxLine) {
+		int[] newArray = new int[maxLine + 1];
+		for (int index = 1; index < maxLine + 1; index++) {
+			newArray[index] = oldArray[index];
+		}
+		return newArray;
+	}
+
+	public int[] makeOffsetArray(CharSequence sourcecode) {
```

Comment:
Also, if you have a plan to make a utility for counting lines in a file.
Please do it in DexterUtil or FileService class